lib/buildpackutil.py

```diff
@@ -433,23 +445,61 @@ def ensure_and_get_mono(mx_version, cache_dir):
     return mono_location
 
 
+def _compose_jdk_url_path(java_version):
```

Comment:
The code in `_compose_jdk_url_path`, `_compose_jvm_target_dir` and `_compose_jre_url_path` are almost identical, but just return slightly different strings. Can we abstract the validation part into a separate function and minimize the amount of copy/pasted code?